modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -198,10 +179,14 @@ public static void setUpClass() throws Exception {
 			HashMapDictionaryBuilder.put(
 				"type", TemplateContextContributor.TYPE_GLOBAL
 			).build());
+
+		_pushServiceContext();
 	}
 
 	@AfterClass
 	public static void tearDownClass() throws Exception {
+		ServiceContextThreadLocal.popServiceContext();
+
 		if (_serviceRegistration != null) {
 			_serviceRegistration.unregister();
 		}
```

Comment:
I would follow the same order from the `setUpClass` method:

```
_serviceRegistration = bundleContext.registerService...

_pushServiceContext();
```

in the `tearDownClass` method:

```
if (_serviceRegistration != null) {
   _serviceRegistration.unregister();
}

ServiceContextThreadLocal.popServiceContext();
```

Reply:
I followed the "stack" concept here. That way, the unregister process will follow the same context it had on the setUpClass.